File: service/app/src/main/java/io/miragon/blueprint/OperatonBikeLeasingApplication.java
package io.miragon.blueprint;

import java.time.Clock;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
public class OperatonBikeLeasingApplication {

    /** Single source of "now" for the app, so time-dependent logic can be pinned in tests. */
    @Bean
    public Clock clock() {
        return Clock.systemDefaultZone();
    }

    public static void main(String[] args) {
        SpringApplication.run(OperatonBikeLeasingApplication.class, args);
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/OperatonBikeLeasingApplication.kt
package io.miragon.blueprint

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import java.time.Clock

@SpringBootApplication
class OperatonBikeLeasingApplication {

    /** Single source of "now" for the app, so time-dependent logic can be pinned in tests. */
    @Bean
    fun clock(): Clock = Clock.systemDefaultZone()
}

fun main(args: Array<String>) {
    runApplication<OperatonBikeLeasingApplication>(*args)
}
